Assets/MixedRealityToolkit.SDK/Inspectors/UX/VisualThemes/ThemeInspector.cs

```diff
@@ -0,0 +1,613 @@
+                    EditorGUILayout.PropertyField(texture, new GUIContent(name, ""), false);
+                    break;
+                case ThemePropertyTypes.Material:
+                    SerializedProperty material = item.FindPropertyRelative("Material");
+                    EditorGUILayout.PropertyField(material, new GUIContent(name, ""), false);
+                    break;
+                case ThemePropertyTypes.AudioClip:
+                    SerializedProperty audio = item.FindPropertyRelative("AudioClip");
+                    EditorGUILayout.PropertyField(audio, new GUIContent(name, ""), false);
+                    break;
+                case ThemePropertyTypes.Animaiton:
+                    SerializedProperty animation = item.FindPropertyRelative("Animation");
+                    EditorGUILayout.PropertyField(animation, new GUIContent(name, ""), false);
+                    break;
+                case ThemePropertyTypes.GameObject:
+                    SerializedProperty gameObjectValue = item.FindPropertyRelative("GameObject");
+                    EditorGUILayout.PropertyField(gameObjectValue, new GUIContent(name, ""), false);
+                    break;
+                case ThemePropertyTypes.String:
+                    stringValue.stringValue = EditorGUILayout.TextField(name, stringValue.stringValue);
+                    break;
+                case ThemePropertyTypes.Bool:
+                    SerializedProperty boolValue = item.FindPropertyRelative("Bool");
+                    boolValue.boolValue = EditorGUILayout.Toggle(name, boolValue.boolValue);
+                    break;
+                case ThemePropertyTypes.AnimatorTrigger:
+                    stringValue.stringValue = EditorGUILayout.TextField(name, stringValue.stringValue);
+                    break;
+                case ThemePropertyTypes.Shader:
+                    SerializedProperty shaderObjectValue = item.FindPropertyRelative("Shader");
+                    EditorGUILayout.PropertyField(shaderObjectValue, new GUIContent(name, ""), false);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        protected bool RenderDeleteButton(int index)
+        {
+            // Create Delete button if we have an array of themes
+            if (themeDefinitions.arraySize > 1 && InspectorUIUtility.SmallButton(RemoveThemePropertyContent))
+            {
+                ClearHistoryCache(index);
+                DeleteThemeDefinition((uint)index);
+
+                serializedObject.Update();
+                EditorUtility.SetDirty(theme);
+                return true;
+            }
+
+            return false;
+        }
+
+        #endregion
+
+        #region Theme Definition Management
+
+        protected virtual void AddThemeDefinition()
+        {
+            Type defaultType = typeof(InteractableActivateTheme);
```

Comment:
Found a bug when adding a new theme via "create and assign theme" button. Got the following stack:

NullReferenceException: Object reference not set to an instance of an object
Microsoft.MixedReality.Toolkit.UI.Editor.ThemeInspector.AddThemeDefinition () (at Assets/MixedRealityToolkit.SDK/Inspectors/UX/VisualThemes/ThemeInspector.cs:450)
Microsoft.MixedReality.Toolkit.UI.Editor.ThemeInspector.OnEnable () (at Assets/MixedRealityToolkit.SDK/Inspectors/UX/VisualThemes/ThemeInspector.cs:42)
UnityEditor.Editor:CreateEditor(Object)
Microsoft.MixedReality.Toolkit.UI.Editor.InteractableInspector:RenderProfileSettings() (at Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs:172)
Microsoft.MixedReality.Toolkit.UI.Editor.InteractableInspector:RenderCustomInspector() (at Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs:94)
Microsoft.MixedReality.Toolkit.UI.Editor.InteractableInspector:OnInspectorGUI() (at Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs:83)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

To repro:

Add second profile
Make the target object a TextMesh object that's not a descendant of the cube where interactable is located.
Try to create a new theme definition

Reply:
This has been fixed